module/zfs/zfs_ioctl.c

```diff
@@ -2984,9 +2985,12 @@ zfs_ioc_set_prop(zfs_cmd_t *zc)
 	 */
 	curtargetip[0] = '\0';
 	(void) nvlist_lookup_string(nvl, ZFS_PROP_TARGET_IP, &targetip);
-	if (targetip != NULL)
+	if (targetip != NULL) {
 		error = zfs_set_targetip_prehook(zc->zc_name, source, targetip,
 		    &curtargetip[0]);
+	} else {
+		error = uzfs_zfs_set_prop(zc->zc_name, source, nvl);
```

Comment:
any errors in zfs_set_prop_nvlist needs reverting the changes done here.. instead, this can be called if no errors after performing zfs_set_prop_nvlist..

Reply:
lets make the fn name as 'uzfs_zinfo_set_prop' as this performs setting few flags on zinfo as well in case of readonly flag..

Reply:
calling function that can set all properties will make those variables get set in zinfo / zv.. for ex., zvol_workers will be set in zv, but, it won't get impacted until zrepl restarts..
So, instead of calling function saying uzfs_zinfo_set_prop, how about setting only for uzfs_zinfo_set_readonly if prop is READONLY

Reply:
done